src/Lecture4.hs

```diff
@@ -134,8 +148,43 @@ errors. We will simply return an optional result here.
 🕯 HINT: Use the 'readMaybe' function from the 'Text.Read' module.
 -}
 
+split :: Char -> String -> [String]
+split symb = go (== symb)
+  where go :: (Char -> Bool) -> String -> [String]
+        go isSymb str = case dropWhile isSymb str of
+                          "" -> []
+                          subStr -> term : go isSymb rest
+                                      where (term, rest) = break isSymb subStr
+
+validateRow :: [String] -> Maybe (String, String, Int)
+validateRow [name, tradeType, cost] = if isNameValid name
```

Comment:
I see that you use the `validate` + `convert` approach. You may be interested in reading about the "Parse, don't validate" approach in Haskell:

* https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/